sshd-common/src/main/java/org/apache/sshd/common/future/DefaultSshFuture.java

```diff
@@ -227,6 +251,13 @@ public void cancel() {
         setValue(CANCELED);
     }
 
+    private boolean isTimeout() {
+        return timeout;
+    }
```

Comment:
Missing empty line between methods. Also make `isTimeout()` public; it may be useful in other contexts. And it must access the flag under lock.